vector/src/main/java/im/vector/riotx/features/home/UnknwonDeviceDetectorSharedViewModel.kt

```diff
@@ -30,26 +31,43 @@ import im.vector.matrix.android.internal.crypto.model.rest.DevicesListResponse
 import im.vector.matrix.rx.rx
 import im.vector.matrix.rx.singleBuilder
 import im.vector.riotx.core.di.HasScreenInjector
-import im.vector.riotx.core.platform.EmptyAction
 import im.vector.riotx.core.platform.EmptyViewEvents
 import im.vector.riotx.core.platform.VectorViewModel
+import im.vector.riotx.core.platform.VectorViewModelAction
+import im.vector.riotx.features.settings.VectorPreferences
+import timber.log.Timber
 import java.util.concurrent.TimeUnit
 
 data class UnknownDevicesState(
         val unknownSessions: Async<List<Pair<MatrixItem?, DeviceInfo>>> = Uninitialized,
         val canCrossSign: Boolean = false
 ) : MvRxState
 
-class UnknownDeviceDetectorSharedViewModel(session: Session, initialState: UnknownDevicesState)
-    : VectorViewModel<UnknownDevicesState, EmptyAction, EmptyViewEvents>(initialState) {
+class UnknownDeviceDetectorSharedViewModel(
+        session: Session,
+        private val vectorPreferences: VectorPreferences,
+        initialState: UnknownDevicesState)
+    : VectorViewModel<UnknownDevicesState, UnknownDeviceDetectorSharedViewModel.Action, EmptyViewEvents>(initialState) {
+
+    sealed class Action : VectorViewModelAction {
+        data class IgnoreDevice(val deviceId: String) : Action()
+    }
+
+    val ignoredDeviceList = ArrayList<String>()
```

Comment:
Maybe a bit useless to keep a local cach of this list. We could only used what's in the sharedPref, there will be no diff regarding perf because there is a cache in the shared pref.